ansible_collections/arista/avd/plugins/plugin_utils/schema/avddataconverter.py

```diff
@@ -175,3 +176,27 @@ def convert_types(self, convert_types: list, data: dict | list, index: str | int
                         return
 
                     yield AvdConversionWarning(key=path_str, oldtype=convert_type, newtype=schema_type)
+
+    def deprecation(self, deprecation: dict, data, schema: dict, path: list):
+        """
+        deprecation:
+          warning: bool, default = True
+          new_key: str
+          removed: bool
+          remove_in_version: str
+          remove_after_date: str
+          url: str
+        """
+
+        deprecation_warning = deprecation.get("warning", True)
+        if not deprecation_warning:
+            return
+
+        yield AvdDeprecationWarning(
+            key=".".join(path),
+            new_key=deprecation.get("new_key"),
+            remove_in_version=deprecation.get("remove_in_version"),
+            remove_after_date=deprecation.get("remove_after_date"),
+            url=deprecation.get("url"),
+            removed=deprecation.get("removed", False),
```

Comment:
It will raise an ansible error in their deprecation tooling. So stop the playbook and say that this key was removed in whatever version and use some other key instead. So more helpful than just removing it from the schema and it will be reported as "unexpected key" by the validator.

Reply:
ok so we would keep on keys in the schema even after their "removal" but that would help error handling when upgrading from quite old versions. is that the goal?

Reply:
I think we can decide on that where we see the need. Now we have the tooling available.